ocaml/xapi/xapi_pool_update.ml

```diff
@@ -446,6 +447,36 @@ let detach_attached_updates __context =
     )
 
 let resync_host ~__context ~host =
+  (* Clean up host_patch table, which is same to Db_gc.single_pass:gc_Host_patches *)
```

Comment:
This code needs attention. I would suggest to work with a local colleague first to make it clearer.

* Binding `get_all` and and `get_record` doesn't add much clarity here.
  If you look at the call sites, you can use the fully qualified names
  there.

* `valid_ref1` and `valid_ref2`: it would be more idiomatic to define
  them as in `let valid_ref1 x = ..... x.host_patch_host`. The names are
  not great. Better names would be: `is_valid_host_patch` and
  `is_valid_pool_patch`.

* Including "ref" into names adds little value.

* This code could be made clearer by using separate let-bindings:

      let table,reference,valid1,valid2 =
        (match DB.get_table_from_ref db (Ref.string_of ref) with
           None -> "UNKNOWN CLASS"
         | Some c -> c),
        (Ref.string_of ref),
        (print_valid ref_1_valid),
        (print_valid ref_2_valid) in

      let table = .. in
      let ref'  = Ref.string_of ref in

  I would do without binding the results of `print_valid ref_1_valid`
  and call these when I need them. To get the necessary space, put
  arguments on their own line:

      debug "resync_host %s (%s) has invalid refs [ref_2: %s; ref_2: %s]. Attempting clean..."
        table
        ref'
        (print_valid ref_1_valid)
        (print_valid ref_2_valid)

* "Attempting clean..." is not using proper grammar. It could be
  "Attempting to clean ..." or "Attempting clean-up".

* I think it is clearer to write:

      if not (is_valid_pool_patch record && is_valid_pool_patch record) then